test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -418,41 +467,82 @@ func (rc *ResourceConsumer) CleanUp() {
-	ginkgo.By(fmt.Sprintf("Running consuming RC %s via %s with %v replicas", name, kind, replicas))
-	_, err := c.CoreV1().Services(ns).Create(context.TODO(), &v1.Service{
+func createService(c clientset.Interface, name, ns string, annotations, selectors map[string]string, port int32, targetPort int) (*v1.Service, error) {
+	return c.CoreV1().Services(ns).Create(context.TODO(), &v1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        name,
-			Annotations: serviceAnnotations,
+			Annotations: annotations,
 		},
 		Spec: v1.ServiceSpec{
 			Ports: []v1.ServicePort{{
 				Port:       port,
 				TargetPort: intstr.FromInt(targetPort),
 			}},
-
-			Selector: map[string]string{
-				"name": name,
-			},
+			Selector: selectors,
 		},
 	}, metav1.CreateOptions{})
+}
+
+// runServiceAndSidecarForResourceConsumer creates service and runs resource consumer for sidecar container
+func runServiceAndSidecarForResourceConsumer(c clientset.Interface, ns, name string, kind schema.GroupVersionKind, replicas int, cpuLimitMillis, memLimitMb int64, podAnnotations, serviceAnnotations map[string]string, additionalContainers []v1.Container, sidecarStatus SidecarStatusType, sidecarType SidecarWorkloadType) {
+	ginkgo.By(fmt.Sprintf("Running consuming RC sidecar %s via %s with %v replicas", name, kind, replicas))
+
+	sidecarName := name + "-sidecar"
+	serviceSelectors := map[string]string{
+		"name": name,
+	}
+	_, err := createService(c, sidecarName, ns, serviceAnnotations, serviceSelectors, port, sidecarTargetPort)
+	framework.ExpectNoError(err)
+
+	ginkgo.By(fmt.Sprintf("Running controller for sidecar"))
+	controllerName := sidecarName + "-ctrl"
+	_, err = createService(c, controllerName, ns, map[string]string{}, map[string]string{"name": controllerName}, port, targetPort)
+	framework.ExpectNoError(err)
+
+	dnsClusterFirst := v1.DNSClusterFirst
+	controllerRcConfig := testutils.RCConfig{
+		Client:    c,
+		Image:     imageutils.GetE2EImage(imageutils.Agnhost),
+		Name:      controllerName,
+		Namespace: ns,
+		Timeout:   timeoutRC,
+		Replicas:  1,
+		Command:   []string{"/agnhost", "resource-consumer-controller", "--consumer-service-name=" + sidecarName, "--consumer-service-namespace=" + ns, "--consumer-port=80"},
+		DNSPolicy: &dnsClusterFirst,
+	}
+
+	framework.ExpectNoError(e2erc.RunRC(controllerRcConfig))
+	// Wait for endpoints to propagate for the controller service.
+	framework.ExpectNoError(framework.WaitForServiceEndpointsNum(
+		c, ns, controllerName, 1, startServiceInterval, startServiceTimeout))
+}
+
+func runServiceAndWorkloadForResourceConsumer(c clientset.Interface, ns, name string, kind schema.GroupVersionKind, replicas int, cpuLimitMillis, memLimitMb int64, podAnnotations, serviceAnnotations map[string]string, additionalContainers []v1.Container, sidecarStatus SidecarStatusType, sidecarType SidecarWorkloadType) {
```

Comment:
`sidecarStatus` and `sidecarType` are unused

Reply:
Removed from argument